rust/src/smb/smb.rs

```diff
@@ -2524,11 +2524,25 @@ pub unsafe extern "C" fn rs_smb_register_parser() {
                 SCLogError!("Invalid max-session-cache-size value");
             }
         }
-        SCLogConfig!("read: max record size: {}, max queued chunks {}, max queued size {}",
-                SMB_CFG_MAX_READ_SIZE, SMB_CFG_MAX_READ_QUEUE_CNT, SMB_CFG_MAX_READ_QUEUE_SIZE);
-        SCLogConfig!("write: max record size: {}, max queued chunks {}, max queued size {}",
-                SMB_CFG_MAX_WRITE_SIZE, SMB_CFG_MAX_WRITE_QUEUE_CNT, SMB_CFG_MAX_WRITE_QUEUE_SIZE);
-        SCLogConfig!("guid: max cache size: {}", SMB_CFG_MAX_GUID_CACHE_SIZE);
+
+        // Suppress static_mut_refs for this debug code. Safe
+        // otherwise.
+        #[allow(unused_lints, static_mut_refs)]
+        {
```

Comment:
would it make sense to contain this stuff in a helper func that can be inlined?

e.g. something like (pseudocode)
```rust
/* SMB_CFG_MAX_READ_SIZE is set at Suricata startup and is static after that... */
#[always_inline]
#[allow(static_mut_refs)]
fn get_max_read_size(...) -> u32 {
    SMB_CFG_MAX_READ_SIZE
}
```

Reply:
that way we can keep the clutter out of the main logic. Esp when addition indents are added things are getting ugly.

Reply:
Good point, and in turn, wrapping the get in a function allows the compiler to reason about it better and you can remove the allow for static_mut_refs.